var/spack/repos/builtin/packages/elsi/package.py

```diff
@@ -33,27 +33,56 @@ class Elsi(CMakePackage):
     variant("enable_sips", default=False, description="Enable SLEPc-SIPs support")
     variant("use_external_elpa", default=False, description="Build ELPA using SPACK")
     variant("use_external_ntpoly", default=False, description="Build NTPoly using SPACK")
-    variant("use_external_omm", default=False, description="Use external libOMM and MatrixSwitch")
     variant("use_external_superlu", default=False, description="Use external SuperLU DIST")
     variant(
         "use_mpi_iallgather", default=True, description="Use non-blocking collective MPI functions"
     )
+    variant(
+        "internal_elpa_version",
+        default="2024",
+        values=("2024", "2023_11", "2023", "2021", "2020"),
+        description="Internal ELPA version",
+        multi=False,
+    )
 
     # Basic dependencies
     depends_on("blas", type="link")
     depends_on("lapack", type="link")
-    depends_on("cmake", type="build")
-    depends_on("mpi")
     depends_on("scalapack", type="link")
+    depends_on("mpi")
 
     # Library dependencies
-    depends_on("elpa", when="+use_external_elpa")
+    with when("+use_external_elpa"):
+        depends_on("elpa+cuda", when="+cuda")
+        depends_on("elpa~cuda", when="~cuda")
     depends_on("ntpoly", when="+use_external_ntpoly")
-    depends_on("slepc", when="+enable_sips")
-    depends_on("petsc", when="+enable_sips")
-    depends_on("superlu-dist", when="+use_external_superlu")
+    with when("+enable_sips"):
+        depends_on("slepc+cuda", when="+cuda")
+        depends_on("slepc~cuda", when="~cuda")
+        depends_on("petsc+cuda", when="+cuda")
+        depends_on("petsc~cuda", when="~cuda")
+    with when("+use_external_superlu"):
+        depends_on("superlu-dist+cuda", when="+cuda")
+        depends_on("superlu-dist~cuda", when="~cuda")
 
     def cmake_args(self):
+        libs_names = ["scalapack", "lapack", "blas"]
+
+        # External libraries
+        if self.spec.satisfies("+use_external_elpa"):
+            libs_names.append("elpa")
+        if self.spec.satisfies("+use_external_ntpoly"):
+            libs_names.append("ntpoly")
+        if self.spec.satisfies("+use_external_superlu"):
+            libs_names.append("superlu-dist")
+
+        lib_paths, libs = [], []
+        for lib in libs_names:
+            lib_paths.extend(
+                [name.replace("-L", "") for name in self.spec[lib].libs.search_flags.split()]
+            )
```

Comment:
```suggestion
            lib_paths.extend(self.spec[lib].libs.directories)
```

Does it make sense? Just for reference an example

```shell
ialberto@daint105:~> spack python
Spack version 0.23.0.dev0
Python 3.6.15, Linux x86_64
>>> from spack.spec import Spec
>>> s = Spec("dla-future")
>>> s.concretize()
==> Warning: The Cray platform, i.e. 'platform=cray', will be removed in Spack v0.23. All Cray machines will be then detected as 'platform=linux'.
>>> s.spec
Traceback (most recent call last):
  File "<console>", line 1, in <module>
AttributeError: 'Spec' object has no attribute 'spec'
>>> s['blas']
intel-oneapi-mkl@=2024.0.0%gcc@=11.2.0~cluster+envmods~ilp64+shared build_system=generic mpi_family=none threads=none arch=cray-cnl7-haswell ^[deptypes=build,link] berkeley-db@=18.1.40%gcc@=11.2.0+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=cray-cnl7-haswell ^[deptypes=build,link,run] bzip2@=1.0.8%gcc@=11.2.0~debug~pic+shared build_system=generic arch=cray-cnl7-haswell ^[deptypes=build] ca-certificates-mozilla@=2023-05-30%gcc@=11.2.0 build_system=generic arch=cray-cnl7-haswell ^[deptypes=build,run] cmake@=3.27.9%gcc@=11.2.0~doc+ncurses+ownlibs build_system=generic build_type=Release arch=cray-cnl7-haswell ^[deptypes=build,link] curl@=8.6.0%gcc@=11.2.0~gssapi~ldap~libidn2~librtmp~libssh~libssh2+nghttp2 build_system=autotools libs=shared,static tls=openssl arch=cray-cnl7-haswell ^[deptypes=build] diffutils@=3.10%gcc@=11.2.0 build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,run] findutils@=4.9.0%gcc@=11.2.0 build_system=autotools patches=440b954 arch=cray-cnl7-haswell ^[deptypes=link] gcc-runtime@=11.2.0%gcc@=11.2.0 build_system=generic arch=cray-cnl7-haswell ^[deptypes=build,link] gdbm@=1.23%gcc@=11.2.0 build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,link] gmake@=4.4.1%gcc@=11.2.0~guile build_system=generic arch=cray-cnl7-haswell ^[deptypes=build,link] hwloc@=2.9.1%gcc@=11.2.0~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=cray-cnl7-haswell ^[deptypes=build,link virtuals=tbb] intel-tbb@=2021.9.0%gcc@=11.2.0~ipo+shared+tm build_system=cmake build_type=Release cxxstd=default generator=make patches=91755c6 arch=cray-cnl7-haswell ^[deptypes=build,link virtuals=iconv] libiconv@=1.17%gcc@=11.2.0 build_system=autotools libs=shared,static arch=cray-cnl7-haswell ^[deptypes=build,link] libpciaccess@=0.17%gcc@=11.2.0 build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,link] libsigsegv@=2.14%gcc@=11.2.0 build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build] libtool@=2.4.7%gcc@=11.2.0 build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,link] libxml2@=2.10.3%gcc@=11.2.0+pic~python+shared build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,run] m4@=1.4.19%gcc@=11.2.0+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=cray-cnl7-haswell ^[deptypes=build,link] ncurses@=6.4%gcc@=11.2.0~symlinks+termlib abi=none build_system=autotools patches=7a351bc arch=cray-cnl7-haswell ^[deptypes=build,link] nghttp2@=1.57.0%gcc@=11.2.0 build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,link] openssl@=3.2.1%gcc@=11.2.0~docs+shared build_system=generic certs=mozilla arch=cray-cnl7-haswell ^[deptypes=build,link,run] perl@=5.38.0%gcc@=11.2.0+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=cray-cnl7-haswell ^[deptypes=build virtuals=pkgconfig] pkg-config@=0.29.2%gcc@=11.2.0+internal_glib build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,link] readline@=8.2%gcc@=11.2.0 build_system=autotools patches=bbf97f1 arch=cray-cnl7-haswell ^[deptypes=build] util-macros@=1.19.3%gcc@=11.2.0 build_system=autotools arch=cray-cnl7-haswell ^[deptypes=build,link,run] xz@=5.4.6%gcc@=11.2.0~pic build_system=autotools libs=shared,static arch=cray-cnl7-haswell ^[deptypes=build,link virtuals=zlib-api] zlib-ng@=2.1.6%gcc@=11.2.0+compat+new_strategies+opt+pic+shared build_system=autotools arch=cray-cnl7-haswell
>>> s['blas'].libs
LibraryList(['/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib/libmkl_gf_lp64.so', '/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib/libmkl_sequential.so', '/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib/libmkl_core.so', '/usr/lib64/libpthread.so', '/usr/lib64/libm.so', '/usr/lib64/libdl.so'])
>>> s['blas'].libs.directories
['/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib', '/usr/lib64']
>>> s['blas'].libs.libraries
['/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib/libmkl_gf_lp64.so', '/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib/libmkl_sequential.so', '/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib/libmkl_core.so', '/usr/lib64/libpthread.so', '/usr/lib64/libm.so', '/usr/lib64/libdl.so']
>>> s['blas'].libs.search_flags
'-L/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib -L/usr/lib64'
>>> s['blas'].libs.directories
['/project/csstaff/ialberto/opt/spack-daint/opt/spack/cray-cnl7-haswell/gcc-11.2.0/intel-oneapi-mkl-2024.0.0-tlbhqpmmewnxlf6fhku3g2kfnjjf4uyc/mkl/2024.0/lib', '/usr/lib64']
```

and some API reference

https://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.FileList
https://spack.readthedocs.io/en/latest/llnl.util.html#llnl.util.filesystem.LibraryList

Reply:
Thank you, much nicer!